config/software/rb-readline.rb

```diff
@@ -17,8 +17,9 @@
 name "rb-readline"
 default_version "master"
 
-dependency "ruby"
-dependency "rubygems"
+# This is only used when we compile our own ruby.
+# Pre-built rubies come with their own readline/libedit.
+dependency "ruby-compiled"
```

Comment:
rb-readline itself is only called on windows.  See ruby.rb.  If someone explicitly wants rb-readline, I am insisting that it be used with a ruby we compile.  This is mostly to break any circular dependency between rb-readline and ruby.  I can completely remove this dependency if you want.